packages/core/admin/admin/src/utils/createRoute.js

```diff
@@ -1,8 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
 import { Route } from 'react-router-dom';
+import { LoadingIndicatorPage } from '@strapi/helper-plugin';
+
+const LazyCompo = ({ loadComponent }) => {
+  const [Compo, setCompo] = useState(null);
+
+  useEffect(() => {
+    const loadCompo = async () => {
+      try {
+        const loadedCompo = await loadComponent();
+
+        setCompo(() => loadedCompo.default);
+      } catch (err) {
+        console.log(err);
+      }
+    };
+
+    loadCompo();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
you can add the "loadCompo" function I think. It will always be referencially identical in our context and will avoid to have this extra lint rule disabled